agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4125,4 +4125,9 @@ std::shared_ptr<bwl::sta_wlan_hal> backhaul_manager::get_selected_backhaul_sta_w
     return (*selected_backhaul_it)->sta_wlan_hal;
 }
 
+bool backhaul_manager::is_main_agent() {
+    auto db = AgentDB::get();
+    return db->device_conf.local_controller && db->device_conf.local_gw;
+}
```

Comment:
IMO this function is redundant. It is more clear to see to seed the flags directly, and a logical AND of two flags does not justify creating a function.